src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/source_filters_table/components/table/table.js

```diff
@@ -0,0 +1,171 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiTableOfRecords,
+  EuiFieldText,
+  keyCodes,
+} from '@elastic/eui';
+
+export class Table extends Component {
+  static propTypes = {
+    indexPattern: PropTypes.object.isRequired,
+    model: PropTypes.shape({
+      data: PropTypes.shape({
+        records: PropTypes.array.isRequired,
+        totalRecordCount: PropTypes.number.isRequired,
+      }).isRequired,
+      criteria: PropTypes.shape({
+        page: PropTypes.shape({
+          index: PropTypes.number.isRequired,
+          size: PropTypes.number.isRequired,
+        }).isRequired,
+        sort: PropTypes.shape({
+          field: PropTypes.string.isRequired,
+          direction: PropTypes.string.isRequired,
+        }).isRequired,
+      }).isRequired,
+    }),
+    deleteFilter: PropTypes.func.isRequired,
+    onDataCriteriaChange: PropTypes.func.isRequired,
+    fieldWildcardMatcher: PropTypes.func.isRequired,
+    saveFilter: PropTypes.func.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      editingFilter: null,
+      editingFilterValue: null,
+    };
+  }
+
+  startEditingFilter = field => this.setState({ editingFilter: field, editingFilterValue: field })
+  stopEditingFilter = () => this.setState({ editingFilter: null })
+  onEditingFilterChange = e => this.setState({ editingFilterValue: e.target.value })
+
+  onEditFieldKeyPressed = ({ charCode }) => {
+    if (keyCodes.ENTER === charCode) {
+      this.props.saveFilter({
+        oldFilterValue: this.state.editingFilter,
+        newFilterValue: this.state.editingFilterValue,
+      });
+      this.stopEditingFilter();
+    }
+  }
+
+  getTableConfig() {
+    const {
+      deleteFilter,
+      onDataCriteriaChange,
+      fieldWildcardMatcher,
+      indexPattern,
+      saveFilter,
+    } = this.props;
+
+    const actions = [];
+
+    if (this.state.editingFilter) {
+      actions.push({
+        name: 'Save',
+        description: 'Save this field',
+        icon: 'checkInCircleFilled',
+        onClick: () => {
+          saveFilter({
+            oldFilterValue: this.state.editingFilter,
+            newFilterValue: this.state.editingFilterValue,
+          });
+          this.stopEditingFilter();
+        }
+      });
+      actions.push({
+        name: 'Cancel',
+        description: 'Cancel editing this field',
+        icon: 'cross',
+        onClick: () => {
+          this.stopEditingFilter();
+        }
+      });
+    }
+    else {
+      actions.push({
+        name: 'Edit',
+        description: 'Edit this field',
+        icon: 'pencil',
+        onClick: filter => this.startEditingFilter(filter.value)
+      });
+      actions.push({
+        name: 'Delete',
+        description: 'Delete this field',
+        icon: 'trash',
+        color: 'danger',
+        onClick: deleteFilter,
+      });
+    }
+
+    return {
+      recordId: 'id',
```

Comment:
I don't think this data has an `id` field. So this makes all the gear icons show up when any part of the table is hovered over, as opposed to just that row's gear icon:

![feb-16-2018 11-27-06](https://user-images.githubusercontent.com/1965714/36325422-9eba791a-130c-11e8-8ca6-8a97e560902f.gif)

Not sure what unique property can be used, since this accepts duplicate `value`s...

Reply:
Nice find!